### PR TITLE
feat: add gitsignsupdatedone autocmd

### PR DESCRIPTION
**Pull request by glepnir**
how do you feel like add an event after update done ?  my situation is my statusline using coroutine and  refresh when event trigger. when the `GitSignsUpdate` triggered the `b:gitsigns_status_dict` may not set or updated so I can't get any useful data from it . add this event will more useful. or move the `GitSignsUpdate` position ?

### CODE REVIEW

**Comment by lewis6991** [comment]
Does the `GitSignsUpdate` not work?

**Comment by glepnir** [comment]
right because this event triggered before variable set or update.  so i can't get any data when it triggered in my stl with coroutine. this pr can work because it trigged after variable set.

**Comment by lewis6991** [comment]
So instead of creating an entirely new event. Just move the old event later.

**Comment by glepnir** [comment]
okay

**Comment by lewis6991** [comment]
Read https://github.com/lewis6991/gitsigns.nvim/blob/main/CONTRIBUTING.md

File: lua/gitsigns/status.lua [hunk 1]
```diff
@@ -25,6 +25,11 @@ function Status:update(bufnr, status)
    vim.b[bufnr].gitsigns_head = status.head or ''
    vim.b[bufnr].gitsigns_status_dict = status
    vim.b[bufnr].gitsigns_status = self.formatter(status)
+
+   api.nvim_exec_autocmds('User', {
+      pattern = 'GitSignsUpdate',
+      modeline = false,
+   })
```
Comment (lewis6991) [inline lua/gitsigns/status.lua:32]: Don't add it here. Keep in it manager.tl

Comment (glepnir) [reply lua/gitsigns/status.lua:32]: done.

File: teal/gitsigns/status.tl [hunk 2]
```diff
@@ -25,6 +25,11 @@ function Status:update(bufnr: integer, status: StatusObj)
   vim.b[bufnr].gitsigns_head = status.head or ''
   vim.b[bufnr].gitsigns_status_dict = status
   vim.b[bufnr].gitsigns_status = self.formatter(status)
+
+   api.nvim_exec_autocmds('User', {
```
Comment (lewis6991) [inline teal/gitsigns/status.tl:29]: This event should not be triggered from this file.

Comment (glepnir) [reply teal/gitsigns/status.tl:29]: after manager.lua 337 ?

Comment (glepnir) [reply teal/gitsigns/status.tl:29]: in there works for me these days. i don't look whole project sry.

**Comment by glepnir** [comment]
emm why closed?

**Comment by lewis6991** [comment]
I'm not submitting this in its current form. I told you changes I wanted and you said.

>  in there works for me these days. i don't look whole project sry.

So I closed the PR.

**Comment by glepnir** [comment]
> in there works for me these days. i don't look whole project sry.

what i mean is triggerd in this file it works fine. I can get the correct data in coroutine.  i don't look  the whole project .just find this update function from `Lspsaga outgoing_calls`.  

>  I told you changes

you don't tell me where is correct pos to put this event ?  you  trigger  before updating  👍  ? 

fine.  is your plugin.

**Pull request closed without merging**